### ISSUE TITLE
lca.switch_method is broken

### ISSUE THREAD

**Issue by BenPortner** [brightway-lca/brightway2-calc#34]
It is currently not possible to switch methods using `lca.switch_method`. The reason: `lca.switch_method` loads `packages` and calls `lca.load_lcia_data(packages)` successfully. However, when calling `lca.lcia()` thereafter, `lca.load_lcia_data()` is again called - this time without arguments. This time `self.packages` is loaded, which still holds the old method data because it was never overwritten during `switch_method`.

### Current implementation:

```python

    def switch_method(self, method):
        """Switch to LCIA method `method`"""
        try:
            _, data_objs, _ = prepare_lca_inputs(method=method)
            packages = [load_package(obj) for obj in data_objs]
        except AssertionError:
            packages = method
        self.method = method
        self.load_lcia_data(packages)


    def lcia(self):
        assert hasattr(self, "inventory"), "Must do lci first"
        # assert self.method, "Must specify a method to perform LCIA"
        if not self.dicts.biosphere:
            raise EmptyBiosphere

        self.load_lcia_data()
        self.lcia_calculation()

```

### Proposed fix:
```python

```python

    def switch_method(self, method):
        """Switch to LCIA method `method`"""
        try:
            _, data_objs, _ = prepare_lca_inputs(method=method)
            packages = [load_package(obj) for obj in data_objs]
        except AssertionError:
            packages = method
        self.method = method
        self.packages = packages
        self.load_lcia_data(packages)


    def lcia(self):
        assert hasattr(self, "inventory"), "Must do lci first"
        # assert self.method, "Must specify a method to perform LCIA"
        if not self.dicts.biosphere:
            raise EmptyBiosphere

        self.load_lcia_data()
        self.lcia_calculation()

```

Additionally, one could introduce a flag to prevent `load_lcia_data` from being called twice.

**Comment by cmutel** [comment]
@BenPortner Thanks, this is a great bug report. All the `switch_X` methods are broken, actually.

I am inclined to fix this by modifying `lcia` to only call `load_lcia_data` if lcia data is not already loaded, similar to how `.normalize` and `.weighting`. Another approach would be to modify [del_resource_group](https://github.com/brightway-lca/bw_processing/blob/master/bw_processing/datapackage.py#L117) from `bw_processing` to allow dropping resources without deleting the underlying resources. This would mean `lca.packages` would still be current after `switch_x`. This is more complicated, but would allow for accessing the packages for e.g. sensitivity analysis.

**Comment by cmutel** [comment]
Note that fix requires `bw_processing` 0.5, released today.
